src/realm/object-store/schema.cpp

```diff
@@ -348,20 +366,42 @@ std::vector<SchemaChange> Schema::compare(Schema const& target_schema, SchemaMod
     return changes;
 }
 
-void Schema::copy_keys_from(realm::Schema const& other) noexcept
+void Schema::copy_keys_from(realm::Schema const& other, bool allow_complete_schema_view) noexcept
 {
-    zip_matching(*this, other, [&](ObjectSchema* existing, const ObjectSchema* other) {
-        if (!existing || !other)
-            return;
-
-        existing->table_key = other->table_key;
-        for (auto& current_prop : other->persisted_properties) {
-            auto target_prop = existing->property_for_name(current_prop.name);
-            if (target_prop) {
-                target_prop->column_key = current_prop.column_key;
-            }
+    auto other_classes = zip_matching(
+        *this, other,
+        [&](ObjectSchema* existing, const ObjectSchema* other) {
+            if (!existing || !other)
+                return;
+            update_or_append_properties(existing, other, allow_complete_schema_view);
+        },
+        allow_complete_schema_view);
+
+    if (!other_classes.empty()) {
+        insert(end(), other_classes.begin(), other_classes.end());
+        sort_schema();
+    }
+}
+
+void Schema::update_or_append_properties(ObjectSchema* existing, const ObjectSchema* other,
+                                         bool allow_complete_schema_view)
+{
+    std::vector<Property> unmatched_properties;
```

Comment:
This temporary isn't doing anything useful and it can just be pushing directly into `existing->persisted_properties`